java/operator/src/main/java/com/instaclustr/cassandra/operator/service/ControllerService.java

```diff
@@ -374,6 +374,24 @@ private void replaceStatefulSet(final DataCenter dataCenter, final V1beta2Statef
             } catch (final Exception e) {
                 logger.warn("Failed to delete PersistentVolume & PersistentVolumeClaim. Reason is: ", e);
             }
+            
+            // waiting for the termination of above pod
+            while (true) {
```

Comment:
Do we want to spin loop on this for ever? Or should we just give up at some point?

Otherwise LGTM

Reply:
Ah, I got what you worried about. If there is a restart of the controller, this [condition](https://github.com/liwang-pivotal/cassandra-operator/blob/2a7c9bf0cc20331e8344b572af7453da38eb56f7/java/operator/src/main/java/com/instaclustr/cassandra/operator/service/ControllerService.java#L383) will be satisfied and then exit the loop. I know the loop here looks ugly but can't find a better way to wait for the finish of scale-down above.